public/src/Helpers.ts

```diff
@@ -28,8 +29,7 @@ export enum ConnStatus {
 const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
 
 /** Returns a string with a prettier format for a deadline */
-export const getFormattedTime = (deadline_string: string): string => {
-    const deadline = new Date(deadline_string)
+export const getFormattedTime = (deadline: Date): string => {
     return `${deadline.getDate()} ${months[deadline.getMonth()]} ${deadline.getFullYear()} ${deadline.getHours()}:${deadline.getMinutes() < 10 ? "0" : ""}${deadline.getMinutes()}`
```

Comment:
Here is a more readable version of this (from #704), please use this:
```ts
const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
const minutes = date.getMinutes()
const zero = minutes < 10 ? "0" : ""
return `${date.getDate()} ${months[date.getMonth()]} ${date.getFullYear()} ${date.getHours()}:${zero}${minutes}`
```
(I also moved the `months` into the func, since it was only used here... not sure if that is smart or not though...)

Reply:
Doubt we need to worry about the following, but I'll mention it nonetheless:

Moving `months` inside the function will cause `months` to be recreated every time the function is called. To avoid this we can create a local closure over the function:

```ts
export const getFormattedTime = (() => {
    const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
    return (deadline: Timestamp) => {
        const date = deadline.toDate()
        const zero = date.getMinutes() < 10 ? "0" : ""
        return `${date} ${months[date.getMonth()]} ${date.getFullYear()} ${date.getHours()}:${zero}${date.getMinutes()}`
    }
})()
```

Where `months` is no longer in the global namespace, nor is it recreated when calling `getFormattedTime`

Reply:
That is interesting.